index.js

```diff
@@ -0,0 +1,161 @@
+const core = require('@actions/core');
+const exec = require('@actions/exec');
+
+const fs = require('fs');
+
+const { parse_bash_array, enable_plugins, move_all } = require('./util.js');
+
+// exec options
+const options = {
+  shell: '/bin/bash'
+};
+
+// Standard out and standard error will be written to these variables
+var myOutput = "";
+var myError = "";
+options.listeners = {
+  stdout: (data) => {
+    myOutput += data.toString();
+  },
+  stderr: (data) => {
+    myError += data.toString();
+  }
+};
+
+// Reading inputs
+const tutor_version = core.getInput('tutor_version');
+const tutor_pearson_plugin_url = core.getInput('tutor_pearson_plugin_url');
+const gh_access_token = core.getInput('gh_access_token');
+const tutor_pearson_plugins = core.getInput('tutor_pearson_plugin_name');
+const tutor_plugin_sources = core.getInput('tutor_plugin_sources');
+const tutor_plugin_names = core.getInput('tutor_plugin_names');
+const extra_private_requirements = core.getBooleanInput('extra_private_requirements');
+const private_repositories = core.getInput('private_repositories');
+const branches = core.getInput('branches');
+const theme_repository  = core.getInput('theme_repository');
+const theme_branch  = core.getInput('theme_branch');
+
+async function run() {
+  try {
+      // Create and activate virtualenv
+      await exec.exec('python3', ['-m', 'venv', 'venv'], options);
+      core.info('Virtualenv created');
+
+      // Install Tutor
+      core.info('Installing Tutor');
+      await exec.exec('venv/bin/python', ['-m', 'pip', 'install', `tutor[full]==${tutor_version}`], options);
+
+      // Install the Tutor Pearson Plugin
+      core.info('Installing Tutor Pearson plugin');
+      const gh_token_url = tutor_pearson_plugin_url.replace("GH_TOKEN", gh_access_token);
+      await exec.exec('venv/bin/pip', ['install', gh_token_url], options);
+
+      // Install Tutor plugins
+      if (tutor_plugin_sources) {
+          core.info('Installing Tutor plugins.');
+          const plugin_sources = parse_bash_array(tutor_plugin_sources);
+          for (var i=0; i < plugin_sources.length; i++) {
+              await exec.exec('venv/bin/pip', ['install',  plugin_sources[i]], options);
```

Comment:
We could reduce the definition of a new variable this way applying the other suggestion. @ShonTitor 
```suggestion
          for (var i=0; i < tutor_plugin_sources.length; i++) {
              await exec.exec('venv/bin/pip', ['install',  tutor_plugin_sources[i]], options);
```